src/streamer.rs

```diff
@@ -224,19 +224,21 @@ fn repair_window(
     //if times flips from all 1s 7 -> 8, 15 -> 16, we retry otherwise return Ok
     if *times & (*times - 1) != 0 {
         trace!("repair_window counter {} {} {}", *times, consumed, received);
-        return Ok(());
     }
 
-    let reqs = find_next_missing(window, crdt, recycler, consumed, received)?;
+    let peers = crdt.read().unwrap().table.len() as u64;
+    let expected = cmp::max(consumed, cmp::max(received, peers) - peers);
```

Comment:
I believe this is equivalent:

```let received = cmp::max(consumed, received.saturating_sub(peers));```

But I'm not really sure because there are no tests on this calculation, on `repair_window`, or `find_next_missing`. Even with that simpler calculation in place, I still don't understand what it means or why both those comparisons are required. I think there needs to be some additional cleanup here. It might be moving this calculation to a separate well-documented function, or tweaking `find_next_missing` to take a parameter with a more precise name than `received`.